### PR TITLE
 Add a section about BOM in the buildpack authors guide

### PR DESCRIPTION
**Pull request by samj1912**
Depends upon #300

### CODE REVIEW

File: content/docs/buildpack-author-guide/create-buildpack/make-buildpack-configurable.md [hunk 1]
```diff
@@ -134,13 +134,8 @@ You will notice that version of Ruby specified in the app's `.ruby-version` file
 [builder] ---> Downloading and extracting Ruby 2.5.0
 ```
 
-Congratulations! You've created your first configurable Cloud Native Buildpack that uses detection, image layers, and caching to create a runnable OCI image.
+Apart from the build logs, we can also store information about the dependencies that our buildpack has provided in the output image for introspection. We will do that in the next section.
```
Comment (samj1912) [inline content/docs/buildpack-author-guide/create-buildpack/make-buildpack-configurable.md:137]: We could add a better follow up here? Not sure if this one is very exciting/correctly worded?

File: content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
++++
+title="Adding Bill-of-Materials"
+weight=409
++++
+
+<!-- test:suite=create-buildpack;weight=9 -->
+
+One of the benefits of buildpacks is they can also populate the app image with metadata from the build process, allowing you to audit the app image for information like  what process types are available and what commands are associated with them, finding the run-image the app image was based on or even finding which buildpacks were used to create the app image.
+
+You can find all of this information using `pack` via its `inspect-image` command.
```
Comment (samj1912) [inline content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md:10]: Any thoughts on the introduction to BOM and pack inspect-image?

I paraphrased this from the rfcs

File: content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+EOL
+
+# 8. ADD A BOM
+cat >> "$layersdir/launch.toml" <<EOL
+[[bom]]
+name = "ruby"
+[bom.metadata]
+version = "$ruby_version"
+EOL
+
```
Comment (samj1912) [inline content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md:128]: Added this minimal BOM to simply replicate what the old BP api would've done before we separated BOM from the build plan. This should help with the concerns that the RFC had about metadata suddenly vanishing.

File: content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+```
+
+Congratulations! You’ve created your first configurable Cloud Native Buildpack that uses detection, image layers, and caching to create an introspectable and runnable OCI image.
+
+## Going further
+
+Now that you've finished your buildpack, how about extending it? Try:
+
+- Caching the downloaded Ruby version
+- Updating the BOM with all the gems provided by bundler
```
Comment (samj1912) [inline content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md:168]: Added this as a follow exercise for the reader to figure it out on their own.

File: content/docs/buildpack-author-guide/create-buildpack/make-buildpack-configurable.md (hunk 1 above)
Comment (dfreilich) [reply content/docs/buildpack-author-guide/create-buildpack/make-buildpack-configurable.md:137]: Hmm. It definitely isn't super exciting, but the other page lead ins aren't super exciting either...
How about something like:
```suggestion
Next, let's see how buildpacks can store information about the dependencies provided in the output image for introspection. 
```

**Review by dfreilich — APPROVED** [review]
This is a fantastic addition @samj1912 . Thanks for putting in the research and work writing this!

**Review by jromero — APPROVED** [review]
superb

**Pull request merged**